### PR TITLE
Fb1369

### PR DESCRIPTION
**Pull request by spiresheep**

### CODE REVIEW

File: Applications/WebPortal/WebApp/library/source/components/drop_down_button.tsx [hunk 1]
```diff
@@ -3,60 +3,61 @@ import { css, StyleSheet } from 'aphrodite/no-important';
 
 interface Properties {
 
-  /* Determines the size element. */
+  /* Determines the size of the element. */
   size?: number|string;
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/components/drop_down_button.tsx]: ```number | string;```

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page.tsx [hunk 2]
```diff
@@ -95,7 +99,7 @@ export class AccountDirectoryPage extends React.Component<Properties, State> {
                 New Account
               </button>
               <div style={AccountDirectoryPage.STYLE.spacing}/>
-              <button onClick={this.props.onNewGroupClick}
+              <button  onClick={this.props.onNewGroupClick}
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page.tsx]: Fix spacing.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page.tsx [hunk 3]
```diff
@@ -206,19 +209,18 @@ export class AccountDirectoryPage extends React.Component<Properties, State> {
       outline: 0,
       borderRadius: 1,
       cursor: 'pointer' as 'pointer',
-      ':active': {
-        backgroundColor: '#4B23A0'
-      },
-      ':focus' : {
+      display: 'flex' as 'flex',
+      flexDirection: 'row' as 'row',
+      alignItems:'center' as 'center',
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/account_directory_page.tsx]: Fix spacing.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import * as Beam from 'beam';
+import { VBoxLayout, Padding } from 'dali';
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: Import in alphabetical order.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 5]
```diff
@@ -27,16 +30,186 @@ interface Properties {
...
+        case(DisplaySize.MEDIUM):
+          return GroupCard.STYLE.accountLabelMedium;
+        case(DisplaySize.LARGE):
+          return GroupCard.STYLE.accountLabelLarge;
+      }
+    })();
+    const accounts = (() => {
+      const accountDetails = [];
+      if(this.props.accounts.length > 0) {
+        for(let i = 0; i < this.props.accounts.length; ++i) {
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: Use ```for(const account of this.props.accounts)```

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 6]
```diff
@@ -27,16 +30,186 @@ interface Properties {
...
+                {this.props.accounts[i].account.name.toString()}
+              </div>
+              <div style={GroupCard.STYLE.rolesBox}>
+                <RolePanel roles={this.props.accounts[i].roles}/>
+              </div>
+            </div>);
+        }
+      } else {
+        accountDetails.push(
+          <div style={{...accountsLableStyle,
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: This needs a unique key too.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 7]
```diff
@@ -27,16 +30,186 @@ interface Properties {
...
+        case(DisplaySize.SMALL):
+          return GroupCard.STYLE.accountLabelSmall;
+        case(DisplaySize.MEDIUM):
+          return GroupCard.STYLE.accountLabelMedium;
+        case(DisplaySize.LARGE):
+          return GroupCard.STYLE.accountLabelLarge;
+      }
+    })();
+    const accounts = (() => {
+      const accountDetails = [];
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: Rename this array to ```accounts```, you can eliminate the IIFE.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 8]
```diff
@@ -27,16 +30,186 @@ interface Properties {
...
+              </VBoxLayout>
+            </div>)}
+        </Transition>
+      </VBoxLayout>);
+  }
+
+  public componentDidUpdate(): void {
+    if(this.props.isOpen &&
+        this.animationStyle.entered.maxHeight
+        !== `${15 + 32 + (32 * this.props.accounts.length)}px`) {
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: Put the ```!==``` operator above, line continuations always end on the operator.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/local_account_directory_model.ts [hunk 9]
```diff
@@ -42,7 +42,10 @@ export class LocalAccountDirectoryModel extends AccountDirectoryModel {
     if(!this.isLoaded) {
       throw Error('Model not loaded.');
     }
-    return this._accounts.get(group).slice();
+    return new Promise<AccountEntry[]>((resolve) => {
+      setTimeout(() => {
+          resolve(this._accounts.get(group).slice());}, 100);
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/local_account_directory_model.ts]: Fix indentation.

**Comment by spiresheep** [comment]
Any other changes to make?

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 10]
```diff
@@ -1,7 +1,11 @@
 import * as Beam from 'beam';
+import { Padding, VBoxLayout } from 'dali';
 import * as React from 'react';
-import { DisplaySize } from '../../display_size';
+import { Transition } from 'react-transition-group';
+import { DisplaySize, DropDownButton, HLine } from '../..';
+import { RolePanel } from '../account_page/role_panel';
 import { AccountEntry } from '.';
+import { any } from 'prop-types';
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: What is this ```any```?

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx [hunk 11]
```diff
@@ -27,16 +31,183 @@ interface Properties {
...
+      switch(this.props.displaySize) {
+        case(DisplaySize.SMALL):
+          return GroupCard.STYLE.accountLabelSmall;
+        case(DisplaySize.MEDIUM):
+          return GroupCard.STYLE.accountLabelMedium;
+        case(DisplaySize.LARGE):
+          return GroupCard.STYLE.accountLabelLarge;
+      }
+    })();
+    let accounts: any[] = [];
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_card.tsx]: Use ```const```.

```accounts``` is of type ```JSX.Element```.

**Pull request merged**